mu-plugins/blocks/local-navigation-bar/src/view.js

```diff
@@ -36,6 +36,64 @@ function init() {
 
 		document.addEventListener( 'scroll', debounce( onScroll ), { passive: true } );
 		onScroll();
+
+		// Check the size of child elements to determine if the local navigation
+		// menu should be collapsed in mobile-view by default. If so, toggle a
+		// CSS class to show the nav block with {"overlayMenu":"always"}
+		// added by `add_extra_navigation`.
+		const onResize = () => {
+			// Bail early on small screens, the visible nav block is already mobile.
+			if ( window.innerWidth <= 600 ) {
+				container.classList.remove( 'wporg-show-collapsed-nav' );
+				return;
+			}
+
+			// Fetch the navWidth from a data value which is set on page load,
+			// so that the uncollapsed visible menu's width is used.
+			let navWidth = container.dataset.navWidth;
+			if ( ! navWidth ) {
+				const navElement = container.querySelector( 'nav:not(.wporg-is-collapsed-nav)' );
+				const navGap = ( window.getComputedStyle( navElement ).gap || '20px' ).replace( 'px', '' ) * 1;
+				// Get the nav width based on items, so that it stays
+				// consistent even if the menu wraps to a new line.
+				const menuItems = navElement.querySelectorAll( '.wp-block-navigation__container > li' );
+				navWidth =
+					[ ...menuItems ].reduce(
+						( acc, current ) => ( acc += current.getBoundingClientRect().width ),
+						0
+					) +
+					navGap * ( menuItems.length - 1 ); // 20px gap between items.
+
+				// Save the value for future resize callbacks.
+				container.dataset.navWidth = Math.ceil( navWidth );
+			}
+
+			const {
+				paddingInlineStart = '0px',
+				paddingInlineEnd = '0px',
+				gap = '0px',
+			} = window.getComputedStyle( container );
+
+			const availableWidth =
+				window.innerWidth -
+				paddingInlineStart.replace( 'px', '' ) * 1 -
+				paddingInlineEnd.replace( 'px', '' ) * 1 -
+				gap.replace( 'px', '' ) * 1;
+
+			const titleElement = container.querySelector( 'p.wp-block-site-title, div.wp-block-group' );
```

Comment:
Noting that `p.wp-block-site-title` is specific and we would be then checking `getBoundingClientRect` on a `null` object.

Reply:
Ah, because we're using that as the heading for showcase mobile… I assumed this would always be a `p` because the page's real `h1` appears in the content (this is true for Developer, Documentation, …everything but Showcase). I've asked about whether Showcase is an exception or another variation https://github.com/WordPress/wporg-mu-plugins/issues/535#issuecomment-1942110550

Reply:
Addressed in https://github.com/WordPress/wporg-mu-plugins/pull/573/commits/9c8edd9ec319bfae2bc8a014cec6e56befbe6928